mibidata/tiff.py

```diff
@@ -343,26 +342,40 @@ def _convert_from_previous(description):
     if not description.get('mibi.fov_name') and description.get(
             'mibi.description'):
         description['mibi.fov_name'] = description.pop('mibi.description')
-    # TODO: Clean up repetition between this and the same MibiImage method
-    if description.get('mibi.folder') and not description.get('mibi.fov_id'):
-        description['mibi.fov_id'] = description['mibi.folder'].split('/')[0]
-        warnings.warn(
-            'The "fov_id" attribute is now required if "folder" is '
-            'specified. Setting "fov_id" to {}.'.format(
-                description['mibi.fov_id']))
-    if (not description.get('mibi.folder') and description.get('mibi.fov_id')
-            and description.get('mibi.fov_id').startswith('FOV')):
-        description['mibi.folder'] = description['mibi.fov_id']
-        warnings.warn(
-            'The "folder" attribute is required if "fov_id" is specified. '
-            'Setting "folder" to {}.'.format(description['mibi.folder']))
+    if description.get('mibi.folder'):
+        description['mibi.fov_id'] = ''
+        value = 'mibi.folder'
```

Comment:
It might be cleaner to just call `mi.MibiImage.match_fov_folder()` in each of these `if` statements. Or can both `if` statements evaluate to True and you need to override the values when `mibi.folder` is present with those for `mibi.fov_id`?

Reply:
I'm not completely sure if there is a case where fov_id and folder are both present in the description but don't match. I was just trying to be on the safe side if somehow the old JSON descriptions didn't enforce this and both if branches execute.